frontend/src/components/Label/Labels.tsx

```diff
@@ -7,6 +7,7 @@ import { KialiIcon } from '../../config/KialiIcon';
 const SHOW_MORE_TRESHOLD = 2;
 
 interface Props {
+  "data-test"?: string;
```

Comment:
Why does this need a `data-test` prop?

Reply:
I'm not sure I like adding this as a prop to the component. It makes it confusing which component the attribute is being added to. e.g.
```
<Dropdown data-test="special-dropdown" />
```
If `Dropdown` had a `data-test` prop then you wouldn't be adding the attribute to the `Dropdown` component which is different from how the rest of the `data-test` attributes work.

Reply:
Just doing `cy.contains('label')` is probably good enough when you are searching for a specific label in a specific part of the screen.

Reply:
> If Dropdown had a data-test prop then you wouldn't be adding the attribute to the Dropdown component itself which is different from how the rest of the data-test attributes work.

Probably the Dropdown is not the best example, as you don't know what element in the DOM will get the `data-test` (it fully depends on Patternfly). But I think I get the idea.

> Just doing cy.contains('label') is probably good enough when you are searching for a specific label in a specific part of the screen.

So, I followed your suggestion and removed this `data-test` prop. It turns out that is not enough to do `cy.contains('label')`. Since this is testing for absence, I need to iterate on all visible labels to test for "not containing" the text. However, I think it is making the test a little bit flaky, because sometimes that label takes a little time to go away, and the assertions may already have ran.

I'm not sure how cypress works when doing `.should('not.exist')` on an element. Perhaps, it waits for it to dissapear??? I'm not sure. It could be that this test has been flaky even when using `.should('not.exist')`.

Reply:
Update: Never mind.

I double checked where this `data-test` prop is used, and now I think it was probably a left-over of some iteration while I was still figuring out what to use on `cy.get` to look for elements. So, I restored the original `.should(not.exist)`.

Yet, I still have that doubt about if it is flaky or not... ugh.